lib/chainweb/p2p/peer.ex

```diff
@@ -0,0 +1,36 @@
+defmodule Kadena.Chainweb.P2P.Peer do
+  @moduledoc """
+  Peer endpoints implementation for P2P API.
+  """
+
+  alias Kadena.Chainweb.P2P.PeerResponse
+  alias Kadena.Chainweb.{Error, Request}
+
+  @type network_opts :: Keyword.t()
+  @type network_id :: :testnet04 | :mainnet01
+  @type error :: {:error, Error.t()}
+  @type location :: String.t()
+  @type retrieve_cut_info_response :: {:ok, PeerResponse.t()} | error()
+
+  @endpoint [cut: "cut/peer", mempool: "mempool/peer"]
```

Comment:
Guys, I think you can avoid the keyword list here, what do you think about this approach? ⬇️ 
```elixir
@cut_endpoint "cut"
@mempool_endpoint "mempool"
@path "peer"

---

:get
|> Request.new(p2p: [endpoint: @cut_endpoint, path: @path])
|> ...
```

Reply:
LGTM